url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/gui/CameraControllerAquynza.java

```diff
@@ -0,0 +1,438 @@
+//===========================================================================
+
+package vsdk.toolkit.gui;
+
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseWheelEvent;
+//import java.awt.event.KeyEvent; // Do not include, deprecated
+
+import vsdk.toolkit.common.linealAlgebra.Vector3D;
+import vsdk.toolkit.common.linealAlgebra.Matrix4x4;
+import vsdk.toolkit.environment.Camera;
+
+public class CameraControllerAquynza extends CameraController {
+
+    private Camera camera;
+    private int oldMouseX;
+    private int oldMouseY;
+    private double deltaMov;
+
+    public CameraControllerAquynza(Camera camera) {
+        this.camera = camera;
+        oldMouseX = 0;
+        oldMouseY = 0;
+        deltaMov = 0.25;
+    }
+
+    public double getDeltaMovement()
+    {
+        return deltaMov;
+    }
+
+    public void setDeltaMovement(double val)
+    {
+        deltaMov = val;
+    }
+
+    public boolean processMouseEventAwt(MouseEvent mouseEvent) {
+      return true;
+    }
+
+    private double augmentLogarithmic(double val, double EPSILON)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.